finality-aleph/src/party.rs

```diff
@@ -1,101 +1,88 @@
 use crate::{
-    environment::Environment, network, network::ConsensusNetwork, AuthorityId, AuthorityKeystore,
-    EpochId, NodeId, SpawnHandle,
+    environment::Environment, network, network::ConsensusNetwork, EpochId, NodeId, SpawnHandle,
 };
 
 use futures::channel::mpsc;
-use rush::{Config as ConsensusConfig, Consensus};
+use rush::Consensus;
 use sc_client_api::backend::Backend;
 use sp_consensus::SelectChain;
 use sp_core::{Blake2Hasher, Hasher, H256};
 use sp_runtime::traits::Block;
-use std::{marker::PhantomData, sync::Arc};
+use futures::SinkExt;
 
-pub(crate) struct ConsensusParty<B, N, C, BE, SC>
-where
-    B: Block,
-    N: network::Network<B>,
-    C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
-    BE: Backend<B> + 'static,
-    SC: SelectChain<B> + 'static,
-{
-    network: ConsensusNetwork<B, H256, N>,
-    client: Arc<C>,
-    select_chain: SC,
-    auth_keystore: AuthorityKeystore,
-    //NOTE: not sure if this phantom is necessary here
-    _phantom: std::marker::PhantomData<BE>,
+pub struct AlephParams<N, C, SC> {
+    pub config: crate::AlephConfig<N, C, SC>,
 }
 
-impl<B, N: network::Network<B>, C, BE, SC> ConsensusParty<B, N, C, BE, SC>
```

Comment:
As mention above, the interface of `ConsensusParty` made it isomorphic to a simple async function, so I replaced it with an async function.

Reply:
You're right, this way is simpler :)